blockchains/eth/lib/filter_errors.js

```diff
@@ -0,0 +1,41 @@
+const trie = require('trie-prefix-tree')
+const { groupBy } = require('lodash')
+
+function setErrors(traces) {
+  let txs = groupBy(traces, tx => tx.transactionHash)
+  const hashes = Object.keys(txs)
+
+  hashes.forEach(hash => {
+    const txTraces = txs[hash]
+
+    const errorTraces = txTraces.filter(trace => typeof trace.error !== 'undefined')
+    if (errorTraces.length === 0) return
+
+    const traceAddresses = txTraces.map(trace => (trace.traceAddress.length > 0) ? '-1 ' + trace.traceAddress.join(' ') : '-1 ')
+    const txTrie = trie(traceAddresses)
+
+    let errAddresses = errorTraces.map(trace => {
+      const errPrefix = '-1 ' + trace.traceAddress.join(' ')
+      const children = txTrie.getPrefix(errPrefix)
+      return children
+    })
+    errAddresses = [].concat(...errAddresses)
+    txs[hash].forEach(trace => {
+      if (errAddresses.indexOf('-1 ' + trace.traceAddress.join(' ')) !== -1) {
+        trace['error'] = 'parent_error'
+      }
+    })
+  })
+  return [].concat(...Object.values(txs))
+
+}
+
+function filterErrors(traces) {
```

Comment:
https://github.com/santiment/blockchain-utils/blob/master/eth/filter_errors.js#L62
Is the `filterNonActions` function removed on purpose? I remember there was some drama and you added that check
https://github.com/santiment/blockchain-utils/commit/9761ff7b11c889aad2942b1ba47294bc4d705558

Reply:
Oh my... I had copied an older version of the file. Thank you for noticing this!